taker/src/main.rs

```diff
@@ -298,6 +318,19 @@ async fn main() -> Result<()> {
 
     let (projection_actor, projection_context) = xtra::Context::new(None);
 
+    let possible_addresses = resolve_maker_addresses(maker_url.as_str()).await?;
+
+    // Assume that the first resolved ipv4 address is good enough for libp2p.
+    let first_maker_address = possible_addresses
+        .iter()
+        .find(|x| x.is_ipv4())
+        .context("Could not resolve maker URL")?;
+
+    let maker_libp2p_address =
+        SocketAddr::new(first_maker_address.ip(), first_maker_address.port() + 1); // libp2p is always the +1 port from legacy networking
```

Comment:
🔧 Perhaps we should encapsulate this knowledge in a function with a docstring.

Reply:
great idea. Both sides should use the same piece of code at all times, it's a shared convention after all.